service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java

```diff
@@ -68,4 +74,15 @@ private Period getPeriod(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
         }
         return encounterParticipantComponents;
     }
+
+    private void getAppointment(Encounter encounter, POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+        Reference referralRequest = encounter.getIncomingReferralFirstRep();
+        Reference patient = encounter.getSubject();
+
+        Optional<Appointment> appointment = appointmentService.retrieveAppointment(referralRequest, patient, clinicalDocument);
+        if (appointment.isPresent()) {
+            encounter.setAppointment(new Reference(appointment.get()));
+            encounter.setAppointmentTarget(appointment.get());
```

Comment:
Isn't it duplicated with EncounterReportBundleService?

Reply:
good point

Reply:
it should be only here int the mapper